### PR TITLE
Implemented magnitude aware sign with test

### PR DESCRIPTION
**Pull request by koenhelwegen**

### CODE REVIEW

File: larq/quantizers.py [hunk 1]
```diff
@@ -61,6 +61,17 @@ def ste_sign(x):
...
 
+@utils.register_keras_custom_object
+def magnitude_aware_sign(x):
+    """
+    Magnitude-aware sign for birealnet.
+    """
+    scale_factor = tf.stop_gradient(
+        tf.reduce_mean(tf.abs(x), axis=list(range(len(x.shape) - 1)))
+    )
+    return tf.math.multiply(scale_factor, ste_sign(x))
```
Comment (lgeiger) [inline larq/quantizers.py]:
```suggestion
    return scale_factor * ste_sign(x)
```

File: larq/quantizers_test.py [hunk 2]
```diff
@@ -69,3 +69,32 @@ def approx_sign_grad(x):
         activation = lq.quantizers.approx_sign(tf_x)
     grad = tape.gradient(activation, tf_x)
     np.testing.assert_allclose(grad.numpy(), approx_sign_grad(x))
+
+
+@pytest.mark.skipif(not tf.executing_eagerly(), reason="requires eager execution")
+def test_magnitude_aware_sign(self):
+    if tf.executing_eagerly():
+        return
```
Comment (lgeiger) [inline larq/quantizers_test.py]: This will not execute.

File: larq/quantizers_test.py [hunk 3]
```diff
@@ -69,3 +69,32 @@ def approx_sign_grad(x):
         activation = lq.quantizers.approx_sign(tf_x)
     grad = tape.gradient(activation, tf_x)
     np.testing.assert_allclose(grad.numpy(), approx_sign_grad(x))
+
+
+@pytest.mark.skipif(not tf.executing_eagerly(), reason="requires eager execution")
+def test_magnitude_aware_sign(self):
```
Comment (lgeiger) [inline larq/quantizers_test.py]:
```suggestion
def test_magnitude_aware_sign():
```

File: larq/quantizers_test.py [hunk 4]
```diff
@@ -69,3 +69,32 @@ def approx_sign_grad(x):
...
+    assert y.shape == x.shape
+
+    # check sign
+    np.testing.assert_allclose(tf.sign(y).numpy(), np.sign(a))
+
+    scale_vector = [np.mean(np.reshape(np.abs(a[:, :, :, i]))) for i in range(3)]
+
+    # check magnitude
+    np.testing.assert_allclose(
+        tf.reduce_mean(tf.abs(y), axis=[0, 1, 2]).numpy(),
```
Comment (lgeiger) [inline larq/quantizers_test.py]:
```suggestion
        lq.quantizers.magnitude_aware_sign(x).numpy(),
```

Shouldn't this test the quantizer?

File: larq/quantizers.py [hunk 5]
```diff
@@ -61,6 +61,17 @@ def ste_sign(x):
     return _binarize_with_identity_grad(x)
 
 
+@utils.register_keras_custom_object
+def magnitude_aware_sign(x):
+    """
+    Magnitude-aware sign for birealnet.
```
Comment (lgeiger) [inline larq/quantizers.py]: It would be great to add arguments, returns and a reference to the docstring

File: larq/quantizers_test.py (hunk 4 above)
Comment (koenhelwegen) [reply larq/quantizers_test.py]: This tests the magnitude of the quantized vector

**Pull request merged**